scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/AzureBlobUploader.java

```diff
@@ -100,29 +106,44 @@ public void init(Map<String, Serializable> args) throws Exception {
     @Override
     public Serializable execute(TaskResult... results) throws IOException, ExecutionException, InterruptedException {
 
-        List<String> filesRelativePathName = new ArrayList<>();
         File file = new File(inputPath);
-
         containerURL = AzureStorageConnectorUtils.createContainerURL(storageAccount, accountKey, containerName);
-
         if (file.exists()) {
-            if (file.isDirectory()) {
-                if (optionalBlobName.isPresent()) {
-                    filesRelativePathName = recursiveFolderUpload(inputPath, optionalBlobName.get(), true);
-                } else {
-                    filesRelativePathName = recursiveFolderUpload(inputPath, "", false);
-                }
+            return (Serializable) uploadResources(file);
+        } else {
+            throw new FileNotFoundException("The input file cannot be found at " + inputPath);
+        }
+    }
 
-            } else {
-                uploadFile(file, optionalBlobName.orElse(file.getName()));
+    /**
+     * This method uploads resources (file or folder to an Azure Blob Storage
+     * @param file
+     * @throws InterruptedException
+     * @throws ExecutionException
+     * @throws IOException
+     */
 
-                filesRelativePathName.add(file.getPath());
+    private List<String> uploadResources(File file) throws InterruptedException, ExecutionException, IOException {
+        List<String> filesRelativePathName = new ArrayList<>();
+        if (file.isDirectory()) {
+            if (optionalBlobName.isPresent()) {
```

Comment:
Is it possible to get rid of "isPresent" ? Maybe to put it as 3rd param of   recursiveFolderUpload().. But how to set the 2nd param without if-else ... To check..